roc/sender-receiver_test.go

```diff
@@ -0,0 +1,151 @@
+	// create context
+	ctx, err := OpenContext(&ContextConfig{})
+	if ctx == nil || err != nil {
+		t.Errorf("Cannot create context, contex: %v, error: %v", ctx, err)
+	}
+
+	// create receiver
+	receiver, err := OpenReceiver(ctx, makeReceiverConfig())
+	if receiver == nil || err != nil {
+		t.Errorf("Cannot create receiver, receiver: %v, error: %v", receiver, err)
+	}
+
+	// bind receiver to two ports
+	// 1)
+	sourceAddr, err := NewAddress(AfAuto, "127.0.0.1", 0)
+	if sourceAddr == nil || err != nil {
+		t.Errorf("Cannot create source address object, address object: %v, error: %v", sourceAddr, err)
+	}
+	err = receiver.Bind(PortAudioSource, ProtoRtpRs8mSource, sourceAddr)
+	if err != nil {
+		t.Errorf("Cannot bind receiver: %v", err)
+	}
+
+	// 2)
+	repairAddr, err := NewAddress(AfAuto, "127.0.0.1", 0)
+	if repairAddr == nil || err != nil {
+		t.Errorf("Cannot create repair address object, address object: %v, error: %v", repairAddr, err)
+	}
+	err = receiver.Bind(PortAudioRepair, ProtoRs8mRepair, repairAddr)
+	if err != nil {
+		t.Errorf("Cannot bind receiver: %v", err)
+	}
+
+	// create sender
+	sender, err := OpenSender(ctx, makeSenderConfig())
+	if sender == nil || err != nil {
+		t.Errorf("Cannot create sender, sender: %v, error: %v", sender, err)
+	}
+
+	// bind sender to a port
+	senderAddr, err := NewAddress(AfAuto, "127.0.0.1", 0)
+	err = sender.Bind(senderAddr)
+	if err != nil {
+		t.Errorf("Cannot bind sender: %v", err)
+	}
+
+	// connect sender to receiver ports
+	// 1)
+	err = sender.Connect(PortAudioSource, ProtoRtpRs8mSource, sourceAddr)
+	if err != nil {
+		t.Errorf("Cannot connect sender to receiver: %v", err)
+	}
+	// 2)
+	err = sender.Connect(PortAudioRepair, ProtoRs8mRepair, repairAddr)
+	if err != nil {
+		t.Errorf("Cannot connect sender to receiver: %v", err)
+	}
+	return sender, receiver, ctx
+}
+
+func Test_roc_sender_write_receiver_read(t *testing.T) {
+	sender, receiver, ctx := prepare(t)
+
+	samplesCnt := 2
+	samples := make([]float32, samplesCnt)
+	for i := 0; i < samplesCnt; i++ {
+		samples[i] = float32(i + 1)
+	}
+
+	endChan := make(chan struct{})
+	go func() {
+		for {
+			err := sender.WriteFloats(samples)
+			if err != nil {
+				t.Fail()
+			}
+			select {
+			case <-endChan:
+				break
+			default:
+			}
+		}
+	}()
+
+	// expectedMatches := 100 * 10 * 4
+	// gotMatches := 0
+	// for {
+	// recFloats := make([]float32, samplesCnt)
+	// err := receiver.ReadFloats(recFloats)
+	// if err != nil {
+	// t.Fail()
+	// }
+	//
+	// nonZeroSamplesCnt := 0
+	// for i := 0; i < samplesCnt; i++ {
+	// if recFloats[i] != 0 {
+	// nonZeroSamplesCnt++
+	// }
+	// }
+	// if nonZeroSamplesCnt == samplesCnt {
+	// gotMatches++
+	// }
+	// if gotMatches >= expectedMatches {
+	// break
+	// }
+	// }
+
+	//endChan <- struct{}{}
+
+	err := receiver.Close()
+	if err != nil {
+		t.Fail()
+	}
+	err = sender.Close()
+	if err != nil {
+		t.Fail()
+	}
+	err = ctx.Close() // remove after finalizers are done
+	if err != nil {
+		t.Fail()
+	}
```

Comment:
Maybe extract this into function too, similar to prepare()?

Also, maybe it makes sense to create a struct ("test fixture") with these triplet (sender, receiver, context).